tests/js/server/aql/aql-optimizer-rule-last-access-on-path.js

```diff
@@ -0,0 +1,189 @@
+        return rules.indexOf(ruleName) !== -1;
+    };
+
+    const assertPlanContainsPathAccess = (nodes) => {
+        assertTrue(planContainsPathAccess(nodes), `We do not have access to path variable p`);
+    };
+    const assertPlanDoesNotContainPathAccess = (nodes) => {
+        assertFalse(planContainsPathAccess(nodes), `We still have access to path variable p`);
+    };
+    const assertRuleIsApplied = (rules) => {
+        assertTrue(ruleIsApplied(rules), `Rule ${ruleName} not found in ${rules}`);
+    };
+    const assertRuleIsNotApplied = (rules) => {
+        assertFalse(ruleIsApplied(rules), `Rule ${ruleName} has been found in ${rules}`);
+    };
+
+    const assertResultsAreNotModifiedByRule = (query) => {
+        const originalResult = db._query(query,{},{optimizer:{rules: [`-${ruleName}`]}}).toArray();
+        const optimizedResult = db._query(query,{},{optimizer:{rules: [`+${ruleName}`]}}).toArray();
+        assertEqual(originalResult, optimizedResult);
+    };
+
+    return {
+        setUpAll: function () {
+            gm._create(graphName, [gm._relation(ec, vc, vc)], [], {numberOfShards: 3});
+            injectData();
+        },
+
+        tearDownAll: function () {
+            gm._drop(graphName, true);
+        },
+
+        testShouldNotImpactNonTraversals: function () {
+            const queriesToTest = [
+                `FOR p IN ${JSON.stringify(hardCodedPaths)} RETURN p.vertices[-1]`,
+                `FOR p IN ${JSON.stringify(hardCodedPaths)} RETURN p.edges[-1]`,
+                `FOR p IN ${vc} RETURN p.vertices[-1]`,
+                `FOR p IN ${vc} RETURN p.edges[-1]`
+            ];
+            for (const q of queriesToTest) {
+                // We need to exclude the move filter into rule, otherwise we cannot simply find the path access anymore
+                const {nodes, rules} = AQL_EXPLAIN(q).plan;
+                assertPlanContainsPathAccess(nodes);
+                assertRuleIsNotApplied(rules);
+
+            }
+        },
+
+        testShouldImpactTraversals: function () {
+            const queriesToTest = [
+                `FOR v, e, p IN 1..2 OUTBOUND "${startVertex}" GRAPH "${graphName}" RETURN p.vertices[-1]`,
+                `FOR v, e, p IN 1..2 OUTBOUND "${startVertex}" GRAPH "${graphName}" RETURN p.edges[-1]`,
+                `FOR v, e, p IN 1..2 OUTBOUND "${startVertex}" ${ec} RETURN p.vertices[-1]`,
+                `FOR v, e, p IN 1..2 OUTBOUND "${startVertex}" ${ec} RETURN p.edges[-1]`,
+                `FOR v, e, p IN 1..2 OUTBOUND "${startVertex}" ${ec} FILTER p.vertices[-1].name == "test" RETURN v`,
+                `FOR v, e, p IN 1..2 OUTBOUND "${startVertex}" ${ec} FILTER p.edges[-1].label == "foo" RETURN e`,
+                `FOR v, e, p IN 1..2 OUTBOUND "${startVertex}" ${ec} FILTER p.edges[-1].label == "foo" && p.vertices[-1].name == "test" RETURN v`,
+                `FOR v, e, p IN 1..2 OUTBOUND "${startVertex}" ${ec} SORT p.edges[-1].timestamp RETURN e`,
```

Comment:
Additional test suggestion:
once we have done the replacement, e.g. replaced `p.vertices[-1]` with `v`, we should also check if projections are still correctly used. For example, `p.vertices[-1].timestamp` will be turned into `v.timestamp`, and in a different optimizer rule the optimizer should be smart enough to figure out that it only needs to extract the `timestamp` attribute from each vertex.
This is a separate optimizer rule, but the combination of the 2 rules should still work, so there should be a small test for that.